validator.go

```diff
@@ -0,0 +1,27 @@
+package fyne
+
+import (
+	"errors"
+	"regexp"
+)
+
+// Validator is a common interface for validating inputs
+type Validator interface {
+	Validate(string) error
+}
+
+// RegexpValidator is a regexp implementation of fyne.Validator
+type RegexpValidator struct {
+	Reason string
+	Regexp *regexp.Regexp
+}
+
```

Comment:
Consider adding `func NewRegexpValidator(regexp, reason string) *RegexpValidator` that can wrap the `regexp.MustCompile`

Reply:
Agreed - except I would not use MustCompile in that case as it will cause panic if errored.
Instead try the compile and log an error if not successful I think? (and don't perform any validation) - better than an app dying :)